apps/api/src/services/email-verification.service.ts

```diff
@@ -0,0 +1,199 @@
+          error: {
+            type: "InvalidToken",
+            token,
+          },
+        };
+      }
+
+      if (tokenRecord.used) {
+        return {
+          success: false,
+          error: {
+            type: "TokenHasBeenUsed",
+            token,
+          },
+        };
+      }
+
+      if (new Date() > tokenRecord.expiresAt) {
+        return {
+          success: false,
+          error: {
+            type: "TokenHasExpired",
+            token,
+          },
+        };
+      }
+
+      await markTokenAsUsed(tokenRecord.id);
+
+      // Check if it's associated with user
+      if (tokenRecord.userId) {
+        const user = await findUserById(tokenRecord.userId);
+        await this.userManager.markEmailAsVerified(tokenRecord.userId);
+
+        // Update Loops with "Verified" userGroup if user exists
+        if (user && user.email && user.name) {
+          await this.updateLoopsWithVerifiedStatus(user.email, user.name);
+        }
+
+        return {
+          success: true,
+          value: {
+            message: "Email verified successfully for user",
+            userId: tokenRecord.userId,
+          },
+        };
+      }
+
+      // Check if it's associated with agent
+      if (tokenRecord.agentId) {
+        const agent = await findAgentById(tokenRecord.agentId);
+        await this.agentManager.markEmailAsVerified(tokenRecord.agentId);
+
+        // Update Loops with "Verified" userGroup if agent exists
+        if (agent && agent.email && agent.name) {
+          await this.updateLoopsWithVerifiedStatus(agent.email, agent.name);
+        }
+
+        return {
+          success: true,
+          value: {
+            message: "Email verified successfully for agent",
+            agentId: tokenRecord.agentId,
+          },
+        };
+      }
+
+      return {
+        success: false,
+        error: {
+          type: "NoAssociation",
+          token,
+        },
+      };
+    } catch (error) {
+      console.error("[EmailVerificationService] Error verifying token:", error);
+      return {
+        success: false,
+        error: {
+          type: "SystemError",
+          message: "Error verifying token",
+          originalError: error,
+        },
+      };
+    }
+  }
+
+  /**
+   * Update Loops with "Verified" userGroup status
+   * @param email The email address to update
+   * @param name The name to update
+   * @returns Promise resolving to the API response, or null if API key is missing
+   */
+  private async updateLoopsWithVerifiedStatus(
+    email: string,
+    name: string,
+  ): Promise<void> {
+    try {
+      // Get the Loops API key from environment variables
+      const loopsApiKey = config.email.apiKey;
+      if (!loopsApiKey) {
+        console.error(
+          "[EmailVerificationService] Missing LOOPS_API_KEY in environment variables",
+        );
+        return;
+      }
+
+      // Prepare the payload for Loops API
+      const loopsPayload = {
```

Comment:
this is fine for now. we'll follow up and align this with the production Loops configuration https://github.com/recallnet/js-recall/issues/773